[2024-06-12T12:05:20.118+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-12T12:05:20.188+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spotify_final_dag.spotify_etl_final manual__2024-06-12T12:04:32.357504+00:00 [queued]>
[2024-06-12T12:05:20.207+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spotify_final_dag.spotify_etl_final manual__2024-06-12T12:04:32.357504+00:00 [queued]>
[2024-06-12T12:05:20.207+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 2
[2024-06-12T12:05:20.235+0000] {taskinstance.py:2330} INFO - Executing <Task(PythonOperator): spotify_etl_final> on 2024-06-12 12:04:32.357504+00:00
[2024-06-12T12:05:20.256+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61: DeprecationWarning: This process (pid=77) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2024-06-12T12:05:20.259+0000] {standard_task_runner.py:63} INFO - Started process 79 to run task
[2024-06-12T12:05:20.269+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'spotify_final_dag', 'spotify_etl_final', 'manual__2024-06-12T12:04:32.357504+00:00', '--job-id', '261', '--raw', '--subdir', 'DAGS_FOLDER/spotify_final_dag.py', '--cfg-path', '/tmp/tmpvge1tndv']
[2024-06-12T12:05:20.271+0000] {standard_task_runner.py:91} INFO - Job 261: Subtask spotify_etl_final
[2024-06-12T12:05:20.366+0000] {task_command.py:426} INFO - Running <TaskInstance: spotify_final_dag.spotify_etl_final manual__2024-06-12T12:04:32.357504+00:00 [running]> on host d317f063dba7
[2024-06-12T12:05:20.613+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_EMAIL='***@example.com' AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spotify_final_dag' AIRFLOW_CTX_TASK_ID='spotify_etl_final' AIRFLOW_CTX_EXECUTION_DATE='2024-06-12T12:04:32.357504+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-06-12T12:04:32.357504+00:00'
[2024-06-12T12:05:20.617+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-12T12:05:20.618+0000] {logging_mixin.py:188} INFO - ETL process started
[2024-06-12T12:05:20.986+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-12T12:05:20.986+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/spotify_final_dag.py", line 130, in ETL
    df = spotify_etl()
         ^^^^^^^^^^^^^
  File "/opt/airflow/dags/spotify_etl.py", line 83, in spotify_etl
    load_df = return_dataframe()
              ^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/spotify_etl.py", line 46, in return_dataframe
    for song in data["items"]:
                ~~~~^^^^^^^^^
KeyError: 'items'
[2024-06-12T12:05:21.006+0000] {taskinstance.py:1206} INFO - Marking task as UP_FOR_RETRY. dag_id=spotify_final_dag, task_id=spotify_etl_final, run_id=manual__2024-06-12T12:04:32.357504+00:00, execution_date=20240612T120432, start_date=20240612T120520, end_date=20240612T120521
[2024-06-12T12:05:21.037+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 261 for task spotify_etl_final ('items'; 79)
[2024-06-12T12:05:21.049+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-06-12T12:05:21.080+0000] {taskinstance.py:3503} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-06-12T12:05:21.083+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
